binding/binding.go

```diff
@@ -101,6 +101,83 @@ func (b *Binding) MatchTopic(exchange string, routingKey string) bool {
 	return b.Exchange == exchange && b.regexp.MatchString(routingKey)
 }
 
+// MatchHeader checks whether the message can be routed on `b` for a
+// header exchange type.
+func (b *Binding) MatchHeader(exchange string, headers *amqp.Table) bool {
+	if b.Exchange != exchange {
+		return false
+	}
+
+	if b.Arguments == nil {
+		if headers == nil {
+			return true
+		}
+		return false
+	}
+
+	if headers == nil {
+		return false
+	}
+
+	bindingArgTable := map[string]interface{}(*b.Arguments)
+	cliHeaders := map[string]interface{}(*headers)
+
+	// @spec-note AMQP 0.9.1
+	//
+	// Any field starting with 'x-' other than 'x-match' is
+	// reserved for future use and will be ignored.
+	//
+	// * 'all' implies that all the other pairs must match the headers
+	// property of a message for that message to be routed (i.e. and AND match)
+	// * 'any' implies that the message should be routed if any of the
+	// fields in the headers property match one of the fields in the
+	// arguments table (i.e. an OR match)
+	//
+	// We arbitrarily choose `all` as the default if none was provided
+	// at binding time.
+	matchType, ok := bindingArgTable["x-match"]
+	if !ok {
+		matchType = "all"
+	}
+
+	allMatch := true
+	anyMatch := false
+
+	for key, value := range bindingArgTable {
```

Comment:
Can u little reorganise that loop, cause 
if matchType == "all" we can break immediately on any negative compare
if matchType == "any" we can break immediately on any positive compare

Reply:
This makes sense, will do